server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -408,12 +419,11 @@ private static boolean upgradeLegacyNodeFolders(Logger logger, Settings settings
                 IOUtils.fsync(nodePath.path, true);
             });
 
-            // now do the actual upgrade. start by upgrading the node metadata file before moving anything, since a downgrade in an
-            // intermediate state would be pretty disastrous
-            loadNodeMetadata(settings, logger, legacyNodeLock.getNodePath());
```

Comment:
This doesn't actually do anything any more, the node metadata is in the persisted cluster state which doesn't get written until much later.